baton/giphy/giphy.go

```diff
@@ -0,0 +1,51 @@
+package giphy
+
+import (
+  "errors"
+  "log"
+  "bytes"
+  "encoding/json"
+  "github.com/gorilla/http"
+)
+type Giphy struct {
+  ApiKey string
+}
+
+func (g Giphy) RunCommand(cmd string, body interface{}, resp chan<- interface{}) error {
+	var url string
+	switch cmd {
+	  case "search":
+	  	query := body.(map[string]interface{})["q"].(string)
+	    url = "http://api.giphy.com/v1/gifs/search?api_key="+g.ApiKey+"&q="+query
```

Comment:
Might using `fmt.Sprintf("http://api.giphy.com/v1/gifs/search?api_key=%s&q=%s", g.ApiKey, query)` be a cleaner way to do this?

Another way to do it might be to have a function that constructs a [URL](https://golang.org/pkg/net/url/) with the `api_key` and `q` parameters (could take them as parameters) and then call the [.String()](https://golang.org/pkg/net/url/#URL.String) method on the returned URL. Thoughts?